### PR TITLE
CVE-2021-28164 - Authorization Before Parsing and Canonicalization in jetty

### PR DESCRIPTION
**Pull request by nrathaus**

### CODE REVIEW

**Comment by DhiyaneshGeek** [comment]
@nrathaus can we move it to CVE-2021-28169 
image

**Comment by nrathaus** [comment]
Oh, sorry, didn't notice it had a CVE - its cool to move to the CVE folder

**Comment by DhiyaneshGeek** [comment]
@nrathaus i went through the other security issues as well

If Possible add this path as well /context/%2e/WEB-INF/web.xml (CVE-2021-28164)

Reference : https://github.com/eclipse/jetty.project/security/advisories/GHSA-v7ff-8wcx-gmc5

**Comment by nrathaus** [comment]
Are you moving it, or should I move it?

File: cves/2021/CVE-2021-28164.yaml [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+    This occurs because both ConcatServlet and WelcomeFilter decode the supplied path to verify it is not within the WEB-INF or META-INF directories. It then uses this decoded path to call RequestDispatcher which will also do decoding of the path. This double decoding allows paths with a doubly encoded WEB-INF to bypass this security check.
+  reference: |
+    - https://github.com/eclipse/jetty.project/security/advisories/GHSA-gwcr-j4wh-j3cq
+    - https://github.com/eclipse/jetty.project/security/advisories/GHSA-v7ff-8wcx-gmc5
+  tags: cve,cve2021,jetty
+
+requests:
+  - method: GET
+    path:
+      - "{{BaseURL}}/concat?/%2557EB-INF/web.xml"
```
Comment (iasdeoupxe) [inline cves/2021/CVE-2021-28164.yaml:18]: This is CVE-2021-28169:

https://github.com/eclipse/jetty.project/security/advisories/GHSA-gwcr-j4wh-j3cq

File: cves/2021/CVE-2021-28164.yaml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+      - "{{BaseURL}}/context/%2e/WEB-INF/web.xml"
```
Comment (iasdeoupxe) [inline cves/2021/CVE-2021-28164.yaml:19]: This is CVE-2021-28164:

https://github.com/eclipse/jetty.project/security/advisories/GHSA-v7ff-8wcx-gmc5

Two different CVEs with two different affected / mixed versions currently get mixed here.

**Comment by ehsandeep** [comment]
Thanks to @DhiyaneshGeek, we can now setup this locally to configure matchers with https://github.com/vulhub/vulhub/tree/master/base/jetty

**Comment by nrathaus** [comment]
I believe this can now be merged

**Pull request closed without merging**